### PR TITLE
Remove tab characters & protect against accidentally introducing them

### PR DESCRIPTION
**Pull request by lschuermann**
### Pull Request Overview

This pull request removes tab characters from Tock's Rust codebase, most of which have been accidentally introduced by me. Whoops, sorry!

While I don't want to "police" anyone's code style, Tock's codebase uses almost exclusively spaces. `rustfmt` seems to replace tabs by spaces pretty consistently, but it appears to have a hard time in macro invocations and some other places. Given that there is no _defined_ standard width for representing tabs in Tock's Rust files, and we're interleaving tabs with spaces, this makes for a very inconsistent indentation depending on the editor's configuration.

Furthermore, this adds a check for tab characters in the codebase to the `run_cargo_fmt.sh` script. It uses `git grep` if available, and falls back to the system's `grep` otherwise. It has been tested with GNU `grep` and might require small modifications for BSD `grep`. The check's overhead isn't noticeable (compared to the `cargo fmt` invocations), even on some low-end systems of mine.

Output prior to removing the tab characters from the codebase:
```
$ make format
Formatting complete.                                        
ERROR: The following files contain tab characters, please use spaces instead:
    -> boards/components/src/led.rs
    -> boards/components/src/led_matrix.rs
    -> capsules/src/l3gd20.rs
    -> capsules/src/lsm303agr.rs
    -> capsules/src/lsm303dlhc.rs
    -> capsules/src/mlx90614.rs
    -> chips/stm32f303xc/src/i2c.rs
    -> kernel/src/debug.rs

Formatting errors.
See above for details
make: *** [Makefile:166: tools/.format_fresh] Error 1
```

### Testing Strategy

This pull request was tested by running the script against the codebase with tab characters, and against the properly formatted one. It requires testing on a machine with BSD grep (such as macOS).

### TODO or Help Wanted

I'd appreciate someone with access to a machine having BSD `grep` to test the fallback to the system's installed `grep`.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/l3gd20.rs [hunk 1]
```diff
@@ -377,9 +377,9 @@ impl Driver for L3gd20Spi<'_> {
     ) -> ReturnCode {
         match subscribe_num {
             0 /* set the one shot callback */ => {
```
Comment (bradjc) [inline capsules/src/l3gd20.rs:379]: Just FYI this is a known rustfmt issue: https://github.com/rust-lang/rustfmt/issues/4119

File: tools/run_cargo_fmt.sh [hunk 2]
```diff
@@ -86,6 +86,15 @@ done
 rm xx00
 printf "\rFormatting complete. %-$((39))s\n" ""
 
+# Check for tab characters in Rust source files that haven't been
+# removed by rustfmt
+RUST_FILES_WITH_TABS="$(git grep --files-with-matches $'\t' -- '*.rs' || grep -lr --include '*.rs' $'\t' || true)"
```
Comment (bradjc) [inline tools/run_cargo_fmt.sh:91]: Unfortunately this gives me `grep: warning: recursive search of stdin` on mac.

**Review by bradjc — COMMENTED** [review]
Good change, just too bad we still have to do this in 2021!

File: tools/run_cargo_fmt.sh (hunk 2 above)
Comment (ppannuto) [reply tools/run_cargo_fmt.sh:91]: Just need to add a directory to start from at the end of the command, which should work on *nix too I believe:

```
$ grep -lr --include '*.rs' $'\t' .
./boards/components/src/led.rs
./boards/components/src/led_matrix.rs
./capsules/src/l3gd20.rs
./capsules/src/lsm303agr.rs
./capsules/src/lsm303dlhc.rs
./capsules/src/mlx90614.rs
./chips/stm32f303xc/src/i2c.rs
./kernel/src/debug.rs
```

**Review by phil-levis — DISMISSED** [review]
Looks good once Brad's comment is addressed.

File: tools/run_cargo_fmt.sh (hunk 2 above)
Comment (lschuermann) [reply tools/run_cargo_fmt.sh:91]: Yeah, that works, can confirm on my new FreeBSD development VM :smile:. Will update accordingly, thanks!

Comment (lschuermann) [reply tools/run_cargo_fmt.sh]: Fixed.

**Review by ppannuto — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**